modules/core/include/opencv2/core/cvdef.h

```diff
@@ -505,7 +503,9 @@ CV_INLINE int cvRound( double value )
 /** @overload */
 CV_INLINE int cvRound(float value)
 {
-#if CV_VFP && !defined HAVE_TEGRA_OPTIMIZATION
+#if defined ANDROID && (defined CV_ICC || defined __GNUC__) && defined HAVE_TEGRA_OPTIMIZATION
```

Comment:
I think it would be better to replace this whole condition with `defined(TEGRA_ROUND_FLT)`. Let O4T itself figure out when to provide it. Ditto for the `double` version.